chain/indexer/src/block.rs

```diff
@@ -151,116 +151,126 @@ where
         };
 
         info!("Building indexer background process");
-        let (tx, rx) = futures::channel::oneshot::channel::<()>();
+        let (tx, mut rx) = futures::channel::mpsc::channel::<()>(1);
 
         spawn(async move {
-            let mut tx = Some(tx);
-
-            let mut block_stream = rpc
-                .try_stream_logs(latest_block_in_db, log_filter)
-                .expect("block stream should be constructible");
+            let is_synced = Arc::new(std::sync::atomic::AtomicBool::new(false));
 
-            let mut events = VecDeque::<Vec<Log>>::new();
             let mut chain_head = 0;
 
-            while let Some(block_with_logs) = block_stream.next().await {
-                info!("Processed block number: {}", block_with_logs.block_id);
-
-                let current_block = block_with_logs.block_id;
-                #[cfg(all(feature = "prometheus", not(test)))]
-                {
-                    METRIC_INDEXER_CURRENT_BLOCK.set(block_with_logs.block_id as f64);
-                }
+            let block_stream = rpc
+                .try_stream_logs(latest_block_in_db, log_filter)
+                .expect("block stream should be constructible")
+                .then(|block_with_logs| {
+                    let rpc = &rpc;
+                    let mut tx = tx.clone();
+                    let is_synced = is_synced.clone();
+
+                    async move {
+                        debug!("Processed block number: {}", block_with_logs.block_id);
```

Comment:
This should be `info!` so we can have some kind of basic logs on the Indexer operation on that level.